core/src/main/java/org/apache/james/core/healthcheck/ResultStatus.java

```diff
@@ -32,4 +32,14 @@
     public String getValue() {
         return value;
     }
+
+    public ResultStatus mergeWith(ResultStatus other) {
```

Comment:
I would propose to have it as a static method:

```
public static ResultStatus merge(ResukltStatus resultStatus1, ResultStatus resultStatus2) {...}
```

This would allow to pass `reduce(ResultStatus::merge)` which is a more symetrical behaviour...